### PR TITLE
feat(accountsdb): add u64 index field to references

### PR DESCRIPTION
**Pull request by 0xNineteen**

### CODE REVIEW

**Comment by 0xNineteen** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/Syndica/sig/384?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#384** <a href="https://app.graphite.dev/github/pr/Syndica/sig/384?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#382** <a href="https://app.graphite.dev/github/pr/Syndica/sig/382?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#381** <a href="https://app.graphite.dev/github/pr/Syndica/sig/381?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @0xNineteen and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by Sobeston — DISMISSED** [review]
Code seems fine, could maybe:
- explain it a bit better in comments (I understand because you previously explained it on slack)
- have some asserts where possible to make sure the next_ fields don't go out of sync

But I'm not going to block you on this.

File: src/accountsdb/index.zig [hunk 1]
```diff
@@ -15,6 +15,8 @@ pub const AccountRef = struct {
     slot: Slot,
     location: AccountLocation,
     next_ptr: ?*AccountRef = null,
+    // NOTE: we store this information to support fast loading
+    next_index: ?u64 = 0,
```
Comment (Sobeston) [inline src/accountsdb/index.zig:19]: Would be great to explain that these fields are in sync (same for AccountReferenceHead).

**Review by Sobeston — CHANGES_REQUESTED** [review]
Just one comment, happy to merge after

File: src/accountsdb/index.zig (hunk 1 above)
Comment (0xNineteen) [reply src/accountsdb/index.zig]: 76800232

**Review by Sobeston — APPROVED** [review]

**Pull request merged**